### PR TITLE
Add folding operator for `IfConfigClauseSyntax`

### PR DESCRIPTION
**Pull request by kimdv**
Resolves https://github.com/apple/sourcekit-lsp/issues/1182

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci please test

File: Tests/SourceKitLSPTests/FoldingRangeTests.swift [hunk 1]
```diff
@@ -357,4 +357,52 @@ final class FoldingRangeTests: XCTestCase {
...
+      ]
+    )
+  }
+
+  func testFoldArgumentsForConditionalElseIfCompileDirectives() async throws {
+    try await assertFoldingRanges(
+      markedSource: """
+        1️⃣#if DEBUG
+          let foo = "x"
+        2️⃣3️⃣#elseif TEST
```
Comment (ahoppen) [inline Tests/SourceKitLSPTests/FoldingRangeTests.swift:379]: There’s not really a reason to specify two markers here. You could just use `2` for both the first range’s end and the next range’s start.

**Comment by kimdv** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test windows

**Comment by kimdv** [comment]
@swift-ci please test windows

**Pull request merged**